app/services/post_alerter.rb

```diff
@@ -37,6 +37,10 @@ def self.create_notification_alert(user:, post:, notification_type:, excerpt: ni
   def self.push_notification(user, payload)
     return if user.do_not_disturb?
 
+    DiscoursePluginRegistry.push_notification_filters.each do |filter|
+      return unless filter.call(user, payload)
```

Comment:
```suggestion
      next unless filter.call(user, payload)
```
You probably don't want `return` here which is the scope above the block.

Reply:
@eviltrout I actually do want to return; this is the mechanism to block notifications from being sent; all filters must pass or the notification is not sent.